### ISSUE TITLE
[BUG]: PCSX2 on linux immediately aborts with assertion failed in function void recRecompile(u32): xGetPtr() - recPtr < _64kb when starting a game but not bios

### ISSUE THREAD

**Issue by yaspoon** [PCSX2/pcsx2#8418]
### Describe the Bug

PCSX2 built from Master on Linux immediately crashes with an abort when trying to launch any game. But the bios is able to boot fine.

This has been happening for a few weeks and I've been `git pull master` a few times but the problem is still around.
I've attached the output from PCSX2 when run which includes the abort line and error line number:
`pcsx2/x86/ix86-32/iR5900-32.cpp:2675: assertion failed in function void recRecompile(u32): xGetPtr() - recPtr < _64kb`

I've also attached the cmake output from the build.
[pcsx2-config.log](https://github.com/PCSX2/pcsx2/files/11007341/pcsx2-config.log)
[pcsx2-crash.log](https://github.com/PCSX2/pcsx2/files/11007342/pcsx2-crash.log)

### Reproduction Steps

Checkout out PCSX2 git with commit `d62a7999fbdc14b56d765b80317bf82b9deb6947`
mkdir build && cd build
cmake ..
make -j24
./bin/pcsx2-qt
Double click "Crash Tag Team racing" in game list, application crashes and closes

### Expected Behavior

The game to launch and be playable

### PCSX2 Revision

v1.7.2175-3361-gd62a7999f

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

Arch

### CPU

intel xeon x5660 * 2

### GPU

Nvidia GTX 1060 6gb driver 525.89.02

### GS Settings

Settings are attached as a screenshot
![Screenshot from 2023-03-18 14-03-49](https://user-images.githubusercontent.com/807332/226081262-f020306f-6503-43e3-888d-1cab08839b86.png)
![Screenshot from 2023-03-18 14-03-55](https://user-images.githubusercontent.com/807332/226081263-277048e8-25c1-40a1-a454-6eaf9e81a281.png)
![Screenshot from 2023-03-18 14-04-00](https://user-images.githubusercontent.com/807332/226081265-cf219d0a-49b9-4627-bdaf-1a652f9ae4e7.png)
![Screenshot from 2023-03-18 14-04-07](https://user-images.githubusercontent.com/807332/226081267-31a557ef-e08c-4ab2-a274-8b721a983a24.png)

### Emulation Settings

Settings are attached as a screenshot. Whether MTVU is enabled or not it still crashes.
![Screenshot from 2023-03-18 14-05-02](https://user-images.githubusercontent.com/807332/226081273-0cfeb44b-ff02-4285-aad2-bf275e6baeea.png)

**Comment by Mrlinkwii** [comment]
Can you try the appimage , looking at the crash log , it dosent like whatever perl version you have installed????

**Comment by kamfretoz** [comment]
AppImage works fine, it looks like a self-compiled native Linux build are borked atm, i have a sneaky suspicion that it has to do with rapidyaml but i will have to dig deeper... i'll keep you posted

**Comment by Mrlinkwii** [comment]
This may be an aur issue , for the time being the appimage should be fine

**Comment by LeCharlatan** [comment]
Did you try to install or reinstall the package "perl-net-dbus" ?

**Comment by kamfretoz** [comment]
Just tested building the native linux version straight from master (NOT from AUR), PCSX2 fail to build with the following error:
```
/home/kamfretoz/stuff/pcsx2/pcsx2/GameDatabase.cpp:180:29: error: no viable conversion from 'c4::yml::ConstNodeRef' to 'const ryml::NodeRef'
for (const ryml::NodeRef& n : node["gameFixes"].children())
```
![image](https://user-images.githubusercontent.com/14798312/226116400-aa94db8b-0500-4562-ae52-96dee769f81d.png)


which seems to suggest that it fails due to `rapidyaml`

the build expect older `rapidyaml` version 0.4.1 present on the system, and will fail to build when compiled against the latest 0.5.1 version

**Comment by TellowKrinkle** [comment]
@kamfretoz can you try building with #8430 and see if that fixes the 0.5 incompatibility?

**Comment by LeCharlatan** [comment]
@TellowKrinkle I may try it if needs

**Comment by LeCharlatan** [comment]
Built a work charm here with #8430 ! :+1:

**Comment by kamfretoz** [comment]
@TellowKrinkle Perfect, #8430 fixes the linux build once again! :)

**Comment by kamfretoz** [comment]
rapidyaml issue aside (sorry for the unintentional hijack btw, my 1 AM brain somehow mistook it as the same issue i was having)

@yaspoon   i can reproduce the error
![Screenshot_20230319_054714](https://user-images.githubusercontent.com/14798312/226143999-0d166063-4e90-476b-8188-2d3c55ed333d.png)

since this is an assertion error, have you tried compiling the release build?
just in case you were unintentionally compiling the devel build.
for the list of cmake preset to use, [check out the PCSX2 building from source guide](https://wiki.pcsx2.net/PCSX2_Documentation/Compiling_on_Linux#Compiling_the_Source)

**Comment by LeCharlatan** [comment]
Hi @kamfretoz
If the problem comes from de devel build, no wonder why it works fine on my build.
I always build PCSX2 using the line "cmake .. -DCMAKE_BUILD_TYPE=Release" in case release isn't the default

**Comment by kamfretoz** [comment]
@LeCharlatan Yup! the issue only manifest on the devel build apparently, which is why it goes unnoticed xD

**Comment by yaspoon** [comment]
> Can you try the appimage , looking at the crash log , it dosent like whatever perl version you have installed????
> 
> Also set your renderer to vulkan and adapter to your Nvidia card

Tried with the vulkan renderer and explicitly setting the adapter to the nvidia card. Same issue. Haven't tried the appimage yet. I am using devel builds and kinda want devel builds because I was hoping to do some performance analysis on the OpenGL renderer and other things.

**Comment by ISuckGood** [comment]
Is not similar to this closed ticket i opened ? #8259
@stenzek  told me not to use debug builds.
